lib/vdc/BOVCollection.cpp

```diff
@@ -0,0 +1,368 @@
+        SetErrMsg(("BOV file missing token  " + _dataSizeToken).c_str());
+        return -1;
+    }
+
+    return 0;
+}
+
+std::string BOVCollection::GetDataFile() const { return _dataFile; }
+
+std::string BOVCollection::GetDataVariableName() const { return _variable; }
+
+std::vector<std::string> BOVCollection::GetSpatialDimensions() const { return _spatialDimensions; }
+
+std::string BOVCollection::GetTimeDimension() const { return _timeDimension; }
+
+std::string BOVCollection::GetUserTime() const { return _time; }
+
+std::vector<size_t> BOVCollection::GetDataSize() const { return _dataSize; }
+
+DC::XType BOVCollection::GetDataFormat() const { return _dataFormat; }
+
+std::vector<float> BOVCollection::GetBrickOrigin() const { return _brickOrigin; }
+
+std::vector<float> BOVCollection::GetBrickSize() const { return _brickSize; }
+
+std::string BOVCollection::GetDataEndian() const { return _dataEndian; }
+
+// Template specialization for reading data of type DC::XType
+template<> int BOVCollection::_findToken<DC::XType>(const std::string &token, std::string &line, DC::XType &value, bool verbose)
+{
+    // Skip comments
+    if (line[0] == '#') { return 0; }
+
+    size_t pos = line.find(token);
+    if (pos != std::string::npos) {    // We found the token
+        std::string format = line;
+        _findTokenValue(format);
+        if (format == _intFormatString)
+            _dataFormat = DC::INT32;
+        else if (format == _floatFormatString)
+            _dataFormat = DC::FLOAT;
+        else if (format == _doubleFormatString)
+            _dataFormat = DC::DOUBLE;
+        else
+            _dataFormat = DC::INVALID;
+
+        if (verbose) { std::cout << std::setw(20) << token << " " << _dataFormat << std::endl; }
+        return 1;
+    }
+    return 0;
+}
+
+// Template specialization for reading data of types bool or string
+template<typename T> int BOVCollection::_findToken(const std::string &token, std::string &line, T &value, bool verbose)
+{
+    // Skip comments
+    if (line[0] == '#') { return 0; }
+
+    size_t pos = line.find(token);
+    if (pos != std::string::npos) {    // We found the token
+        _findTokenValue(line);
+        stringstream ss(line);
+        if (std::is_same<T, bool>::value) {
+            ss >> std::boolalpha >> value;
+        } else {
+            ss >> value;
+        }
+
+        if (verbose) { std::cout << std::setw(20) << token << " " << value << std::endl; }
+
+        if (ss.eof() == 0) {
+            std::string message = "The keyword " + token + " may only contain one value.";
+            SetErrMsg(message.c_str());
+            return -1;    // Bad token
+        }
+
+        if (ss.bad()) {
+            std::string message = "Invalid value for " + token + " in BOV header file.";
+            SetErrMsg(message.c_str());
+            return -1;    // Bad token
+        }
+        return 1;    // we now have a token
+    }
+    return 0;    // we didn't find the token
```

Comment:
This function has a 3 state return so it would be clearer if the integer values were replaced with a success/error/not_found enum.

Reply:
Interesting.  Fixed.